fastparquet/util.py

```diff
@@ -106,3 +122,21 @@ def index_like(index):
                 index._start == 0 and
                 index._stop == len(index) and
                 index._step == 1 and index.name is None)
+
+
+def check_column_names(columns, *args):
+    """Ensure that parameters listing column names have corresponding columns"""
+    for arg in args:
+        if isinstance(arg, (tuple, list)):
+            if set(arg) - set(columns):
+                raise ValueError("Column name not in list.\n"
+                                 "Requested %s\n"
+                                 "Allowed %s" % (arg, columns))
+
+
+def byte_buffer(raw_bytes):
+    return bytearray(raw_bytes) if PY2 else memoryview(raw_bytes)
```

Comment:
The motivation for this is a bit unclear, or at least the semantics are a bit vague. `memoryview` produces a (possible writable) view, while `bytearray` produces a mutable copy.

If this is about `np.frombuffer`, then it's true that it doesn't accept memoryview on Python 2, but OTOH it seems to accept the old-style `buffer` object.